app/models/facilities_query.rb

```diff
@@ -0,0 +1,178 @@
+# frozen_string_literal: true
+
+class FacilitiesQuery
+  attr_reader :params
+
+  def initialize(params)
+    @params = params
+  end
+
+  def query
+    if valid_location_query?
+      location_query_klass.new(params).run
+    elsif @params[:ids]
+      build_result_set_from_ids(params[:ids]).flatten
+    else
+      BaseFacility.none
+    end
+  end
+
+  # Massages the location_query_klass return value to be boolean
+  def valid_location_query?
+    !location_query_klass.nil?
+  end
+
+  # When given more than one type of distance query param,
+  # return nil because the app will not choose preference for the user.
+  # In the controller, this logic is somewhat duplicated
+  # and will render an error when given multiple params.
+  # In the case that only one of these types is given,
+  # return the class used to make that type of query.
+  def location_query_klass
+    @location_query_klass ||= case location_keys
+                              when [] then nil
```

Comment:
this could be probably be left out as a case since no match would return nil (I think) .. probably more clear to leave if this is an expected value though

Reply:
I like erring on the side of being explicit when the case is expected.

Reply:
Not particularly. This logic had been changed like 20 times, and the original "else" statement threw an error, so that's why there was a distinction. But the `when [] then nil` line isn't necessary anymore